packages/core/src/actions/stop.ts

```diff
@@ -31,21 +36,40 @@ export function stop<
   const actor = actorRef;
 
   return createDynamicAction(
-    stopActionType,
     {
-      actor
+      type: stopActionType,
+      params: {
+        actor
+      }
     },
-    ({ params, type }, context, _event, { state }) => {
```

Comment:
removal of `params` from here feels a little bit clunky at times since we need to reach into the closure and at times we end up duplicating things like here:
https://github.com/statelyai/xstate/blob/7106e9daa44cf32891a34a2cdce75881c3048349/packages/core/src/actions/send.ts#L58-L81

If we'd bring params back as a parameter here then we could even move those final "action handlers" out of the factories.
```ts
// pre
const assign = (assignment) => {
  return createDynamicAction(
    { type: "assign", params: { assignment } },
    (event, { state }) => {
      //...
    }
  );
};

// post
const assignHandler = (event, { state }) => {
  //...
};

const assign = (assignment) => {
  return createDynamicAction(
    { type: "assign", params: { assignment } },
    assignHandler
  );
};
```